go/libkb/appstate.go

```diff
@@ -85,6 +85,57 @@ func (a *MobileAppState) StateAndMtime() (keybase1.MobileAppState, *time.Time) {
 
 // --------------------------------------------------
 
+// MobileNetState tracks the state of the network status of the app in which
+// the service is running in.
+type MobileNetState struct {
+	Contextified
+	sync.Mutex
+	state     keybase1.MobileNetworkState
+	updateChs []chan keybase1.MobileNetworkState
+}
+
+func NewMobileNetState(g *GlobalContext) *MobileNetState {
+	return &MobileNetState{
+		Contextified: NewContextified(g),
+		state:        keybase1.MobileNetworkState_NONE,
+	}
+}
+
+// NextUpdate returns a channel that triggers when the network state changes
+func (a *MobileNetState) NextUpdate(lastState *keybase1.MobileNetworkState) chan keybase1.MobileNetworkState {
+	a.Lock()
+	defer a.Unlock()
+	ch := make(chan keybase1.MobileNetworkState, 1)
+	if lastState != nil && *lastState != a.state {
+		ch <- a.state
+	} else {
+		a.updateChs = append(a.updateChs, ch)
+	}
+	return ch
+}
+
+// Update updates the current network state, and notifies any waiting calls
+// from NextUpdate
+func (a *MobileNetState) Update(state keybase1.MobileNetworkState) {
```

Comment:
This guy should just have a `State` function like the app state thing that inspires it.